AirCasting/MicrophoneSession/MicrophoneManager.swift

```diff
@@ -21,86 +18,110 @@ class MicrophoneManager: ObservableObject {
+            ]
+            
+            do {
+                try audioSession.setCategory(AVAudioSession.Category.playAndRecord)
+                try audioSession.setActive(true)
+                try recorder = AVAudioRecorder(url:url, settings: recordSettings)
+                
+            } catch {
+                return
             }
-        }
-        
-        self.session = session
-        createDBStream()
-        locationProvider.requestLocation()
-        
-        let url = URL(fileURLWithPath: "/dev/null", isDirectory: true)
-        
-        let recordSettings: [String: Any] = [
-            AVFormatIDKey: NSNumber(value: kAudioFormatAppleLossless),
-            AVSampleRateKey: 44100.0,
-            AVNumberOfChannelsKey: 1,
-            AVEncoderAudioQualityKey: AVAudioQuality.min.rawValue
-        ]
-        
-        do {
-            try audioSession.setCategory(AVAudioSession.Category.playAndRecord)
-            try audioSession.setActive(true)
-            try recorder = AVAudioRecorder(url:url, settings: recordSettings)
             
-        } catch {
-            return
+            recorder.isMeteringEnabled = true
+            recorder.record()
+            isRecording = true
+            
+            levelTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(getMeasurement), userInfo: nil, repeats: true)
         }
-        
-        recorder.isMeteringEnabled = true
-        recorder.record()
-        isRecording = true
-        
-        levelTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(getMeasurement), userInfo: nil, repeats: true)
     }
     
     @objc func getMeasurement() {
+        if !recorder.isRecording {
+            recorder.record()
+        }
         recorder.updateMeters()
-        
         let level = recorder.averagePower(forChannel: 0)
+        createMeasurement(value: level)
+    }
+    
+    func stopRecording() {
+        levelTimer.invalidate()
+        isRecording = false
+        recorder.stop()
+        recorder = nil
+        session?.status = Int16(SessionStatus.FINISHED.rawValue)
+        do {
+            try context.save()
+        } catch {
+            print("Error for saving session when stopping mic recording: \(error)")
+        }
+    }
+    
+    @objc func handleInterruption(notification: Notification) {
+        guard let userInfo = notification.userInfo,
+            let typeValue = userInfo[AVAudioSessionInterruptionTypeKey] as? UInt,
+            let type = AVAudioSession.InterruptionType(rawValue: typeValue) else {
+                return
+        }
+
+        switch type {
+        case .began:
+            levelTimer.invalidate()
+            session?.status = Int16(SessionStatus.DISCONNETCED.rawValue)
+        case .ended:
+            levelTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(getMeasurement), userInfo: nil, repeats: true)
+        default: ()
```

Comment:
Please use all the cases rather then using default. Let the compilation help us and if new case would be added a failure here would require to rethink this situation, how does the new enum influence the flow. Without it may be forgotten and ignored and that could lead to potential bugs that would be very hard to find.